### PR TITLE
Dev/mzielins/add table generator

### PR DESCRIPTION
**Pull request by marziel2001**
Pushed just to show the progress. Any feedback is appreciated.

### CODE REVIEW

File: react-app/src/pages/TablePage.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { useState } from "react"
+
+export default function TablePage() {
+    const [TableContent, setContent] = useState<[number, string, number][]>([[0, "Marcel", 130]])
```
Comment (RZEPI) [inline react-app/src/pages/TablePage.tsx]: You could use a type definition from TypeScript, I also suggest to make it object not a list for better readability

File: react-app/src/styles/TablePage.module.css
Comment (RZEPI) [inline react-app/src/styles/TablePage.module.css:1]: Could you adjust the styling to current page appearance?
We could also make a table's body scrollable, but this is just my preference

Comment (marziel2001) [reply react-app/src/styles/TablePage.module.css:1]: Styling adjusted. I will think about separate scroll for the table.

File: react-app/src/pages/TablePage.tsx (hunk 1 above)
Comment (marziel2001) [reply react-app/src/pages/TablePage.tsx]: Good point about making it an object. Will address it tomorrow.

File: react-app/src/App.tsx [hunk 2]
```diff
@@ -11,6 +14,12 @@ const router = createBrowserRouter([
     children: [{ index: true, element: <HomePage /> }],
     errorElement: <ErrorPage />,
   },
+  {
+    path: "/TablePage",
+    element: <RootLayout />,
+    children: [{ index: true, element: <TablePage /> }],
+    errorElement: <ErrorPage />,
+  },
```
Comment (RZEPI) [inline react-app/src/App.tsx:17]: It's better to add this path as a children of main route

File: react-app/src/components/UI/TableRow.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  t,
+  className,
+}: {
+  t: TableRowData;
+  className: string;
+}) {
+  const [focus, setFocus] = useState<boolean>(false);
+
+  function onClick() {
+    setFocus(!focus);
```
Comment (RZEPI) [inline react-app/src/components/UI/TableRow.tsx]: setState function shouldn't use it's state old value, instead it should change a the prevState in lambda function, for example: setFocus(prevFocus => !prevFocus)

File: react-app/src/pages/TablePage.tsx [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+      t={tuple}
+      key={tuple.id}
+    ></TableRow>
+  ));
+
+  function addNRecords(n: number) {
+    let NameIndex: number;
+    let LevelIndex: number;
+
+    const TmpArray: TableRowData[] = [];
```
Comment (RZEPI) [inline react-app/src/pages/TablePage.tsx]: wrong constatnt and variables case

File: react-app/src/pages/TablePage.tsx [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+      }
+    );
+    }
+
+    setContent([...TmpArray, ...TableContent]);
+  }
+
+  function deleteRecord() {
+    TableContent.splice(0, 1);
+    setContent([...TableContent]);
```
Comment (RZEPI) [inline react-app/src/pages/TablePage.tsx]: we should probably work with a copy of the state here to avoid unpredicted behaviour

**Review by RZEPI — CHANGES_REQUESTED** [review]

File: react-app/src/pages/TablePage.tsx [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+    }
+
+    setContent(tmpArray);
+  }
+
+  function swapRows() {
+    const tmpArray: TableRowData[] = [...tableContent];
+
+    const Index1 = Math.floor(Math.random() * tableContent.length);
+    const Index2 = Math.floor(Math.random() * tableContent.length);
```
Comment (RZEPI) [inline react-app/src/pages/TablePage.tsx]: PascalCase instead of camelCase

**Review by RZEPI — CHANGES_REQUESTED** [review]

**Review by RZEPI — APPROVED** [review]

**Review by NOTSAHLO — APPROVED** [review]

**Pull request merged**